x-pack/plugin/inference/qa/inference-service-tests/src/javaRestTest/java/org/elasticsearch/xpack/inference/MockInferenceServiceIT.java

```diff
@@ -63,4 +64,42 @@ public void testMockService_DoesNotReturnSecretsInGetResponse() throws IOExcepti
         assertNull(putServiceSettings.get("api_key"));
         assertNotNull(putServiceSettings.get("model"));
     }
+
+    @SuppressWarnings("unchecked")
+    public void testMockService_DoesNotReturnHiddenField_InModelResponses() throws IOException {
+        String inferenceEntityId = "test-mock";
+        var putModel = putModel(inferenceEntityId, mockServiceModelConfig(), TaskType.SPARSE_EMBEDDING);
+        var getModels = getModels(inferenceEntityId, TaskType.SPARSE_EMBEDDING);
+        var model = ((List<Map<String, Object>>) getModels.get("models")).get(0);
+
+        for (var modelMap : List.of(putModel, model)) {
+            assertEquals(inferenceEntityId, modelMap.get("model_id"));
+            assertThat(modelMap.get("service_settings"), is(Map.of("model", "my_model")));
+            assertEquals(TaskType.SPARSE_EMBEDDING, TaskType.fromString((String) modelMap.get("task_type")));
+            assertEquals("test_service", modelMap.get("service"));
+        }
+
+        // The response is randomly generated, the input can be anything
+        var inference = inferOnMockService(inferenceEntityId, List.of(randomAlphaOfLength(10)));
+        assertNonEmptyInferenceResults(inference, 1, TaskType.SPARSE_EMBEDDING);
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testMockService_DoesReturnHiddenField_InModelResponses() throws IOException {
+        String inferenceEntityId = "test-mock";
+        var putModel = putModel(inferenceEntityId, mockServiceModelConfig(null, true), TaskType.SPARSE_EMBEDDING);
+        var getModels = getModels(inferenceEntityId, TaskType.SPARSE_EMBEDDING);
+        var model = ((List<Map<String, Object>>) getModels.get("models")).get(0);
+
+        for (var modelMap : List.of(putModel, model)) {
+            assertEquals(inferenceEntityId, modelMap.get("model_id"));
+            assertThat(modelMap.get("service_settings"), is(Map.of("model", "my_model", "hidden_field", "my_hidden_value")));
```

Comment:
Since this test passes `true` for `should_return_hidden_field` we should see it in the response